pkg/controllers/subnetset/subnetset_controller.go

```diff
@@ -111,11 +111,18 @@ func (r *SubnetSetReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 	} else {
 		if controllerutil.ContainsFinalizer(obj, servicecommon.SubnetSetFinalizerName) {
 			metrics.CounterInc(r.SubnetService.NSXConfig, metrics.ControllerDeleteTotal, MetricResTypeSubnetSet)
-			if err := r.DeleteSubnetForSubnetSet(*obj, false); err != nil {
+			hasStaleSubnetPorts, err := r.DeleteSubnetForSubnetSet(*obj, false)
+			if err != nil {
 				log.Error(err, "deletion failed, would retry exponentially", "subnetset", req.NamespacedName)
 				deleteFail(r, &ctx, obj, "")
 				return ResultRequeue, err
 			}
+			if hasStaleSubnetPorts {
+				err := fmt.Errorf("stale subnet ports found while deleting subnetset %v", req.NamespacedName)
+				log.Error(err, "deletion failed, delete all the subnetports first", "subnetset", req.NamespacedName)
+				updateFail(r, &ctx, obj, err.Error())
```

Comment:
updateFail will only update subnetset CR status not Requeue, we should requeue it to make controller retry to delete subnets.

Reply:
ok